src/circuits/circuits.jl

```diff
@@ -105,16 +105,20 @@ appendlayer!(gates::AbstractVector{ <: Tuple},
 
 Layer of two-qubit gates 
 """
-function twoqubitlayer(gatename::String,bonds::Array)
+function twoqubitlayer(gatename::String,bonds::Array; rng = Random.GLOBAL_RNG)
```

Comment:
The current design of this function feels a bit off. It seems to bake in the assumption that `"randU"` is the only gate with randomness. Also, if gates have other parameters, how do they get passed (for example a gate like `"XX"` that accepts angles)? It seems like how it should be designed is to accept general keyword arguments, which then get passed to each gate. For example:
```julia
function twoqubitlayer(gatename::String, bonds::Array; kwargs...)
  gates = Tuple[]
  for bond in bonds
    # I forget if this is the right way to pass kwargs to the gates as a NamedTuple, but you get the idea
    push!(gates, (gatename, bond, kwargs)) 
  end
return gates
end
```
Then, `"randU"` should accept an RNG as a parameter (instead of the entire gate), where the default is `Random.GLOBAL_RNG`. Then, the syntax for making a layer with its own random stream would be the same as the current version:
```julia
rng = MersenneTwister(1234)
twoqubitlayer("randU", [(1, 2), (3, 4)]; rng = rng)
```
but the `rng` would get passed straight to the `"randU"` gate (with the bonus that other parameters could be passed as well).

Reply:
Also, on another note, this functions name seems to be too specific for its functionality, for example it seems like it would work for the input:
```julia
twoqubitlayer("H", [1, 2, 3, 4])
```
and also for 3-qubit gates if a vector of `Tuple{Int, Int, Int}` was passed. Should this be renamed something like `uniformgatelayer` or just `gatelayer`?